File: app/src/main/java/com/example/mvctutorial/designpattern/_12visitor/VisitorA.java
package com.example.mvctutorial.designpattern._12visitor;

public class VisitorA implements Visitor {
    private int number;

    public VisitorA() {
        this.number = 0;
    }

    @Override
    public void visit(Visitable visitable) {
        if (visitable instanceof VisitableA) {
            number += ((VisitableA) visitable).numberOfMember;
        }
    }

    public int getNumber() {
        return number;
    }

    public void setNumber(int number) {
        this.number = number;
    }
}

File: app/src/main/java/com/example/mvctutorial/designpattern/_12visitor/VisitorA.kt
package com.example.mvctutorial.designpattern._12visitor

class VisitorA(var number: Int = 0): Visitor {
    override fun visit(visitable: Visitable) {

        if (visitable is VisitableA){
            number += visitable.numberOfMember

        }
    }
}